cli/azd/pkg/project/framework_service_docker.go

```diff
@@ -34,15 +34,15 @@ import (
 )
 
 type DockerProjectOptions struct {
-	Path        string                  `yaml:"path,omitempty"        json:"path,omitempty"`
-	Context     string                  `yaml:"context,omitempty"     json:"context,omitempty"`
-	Platform    string                  `yaml:"platform,omitempty"    json:"platform,omitempty"`
-	Target      string                  `yaml:"target,omitempty"      json:"target,omitempty"`
-	Registry    osutil.ExpandableString `yaml:"registry,omitempty"    json:"registry,omitempty"`
-	Image       osutil.ExpandableString `yaml:"image,omitempty"       json:"image,omitempty"`
-	Tag         osutil.ExpandableString `yaml:"tag,omitempty"         json:"tag,omitempty"`
-	RemoteBuild bool                    `yaml:"remoteBuild,omitempty" json:"remoteBuild,omitempty"`
-	BuildArgs   []string                `yaml:"buildArgs,omitempty"   json:"buildArgs,omitempty"`
+	Path        string                    `yaml:"path,omitempty"        json:"path,omitempty"`
+	Context     string                    `yaml:"context,omitempty"     json:"context,omitempty"`
+	Platform    string                    `yaml:"platform,omitempty"    json:"platform,omitempty"`
+	Target      string                    `yaml:"target,omitempty"      json:"target,omitempty"`
+	Registry    osutil.ExpandableString   `yaml:"registry,omitempty"    json:"registry,omitempty"`
+	Image       osutil.ExpandableString   `yaml:"image,omitempty"       json:"image,omitempty"`
+	Tag         osutil.ExpandableString   `yaml:"tag,omitempty"         json:"tag,omitempty"`
+	RemoteBuild bool                      `yaml:"remoteBuild,omitempty" json:"remoteBuild,omitempty"`
+	BuildArgs   []osutil.ExpandableString `yaml:"buildArgs,omitempty"   json:"buildArgs,omitempty"`
```

Comment:
Docker users might want to map the build arguments to env vars using docker directly (see: https://docs.docker.com/reference/cli/docker/buildx/build/#build-arg) 

When that's te case, azd should either use the full system environment when calling docker, or check if there are any buildArgs like `FOO_VAR` (w/o using the `=` symbol) and grab `FOO_VAR` from system env and use it when calling `docker`.

Having expandable string support is a plus, but it might not be idiomatic/expected to docker users.